### ISSUE TITLE
Plugin should work with Unix-style file path on Windows

### ISSUE THREAD

**Issue by php-coder** [jarslab/babel-maven-plugin#7]
When I specified

> <sourceDir>${basedir}/src/main/javascript</sourceDir>

It didn't work on Windows and always created resulted file in the source dir instead of the targetDir.

It happends because the following code from TargetFileWriter:

```java
        final String targetDirectory = sourceFile.toString()
                .replace(sourceFileName.toString(), "")
                .replace(sourceDir, targetDir);
```
tries to remove "C:\Users\slava\git\mystamps/src/main/javascript" from string "C:\Users\slava\git\mystamps\src\main\javascript\components\" and it couldn't do that.

As you can see, the slashes (/ vs \) are different.

It would be good if plugin will handle paths with any type of the directories separators similarly.

The current workaround is:
> <sourceDir>${basedir}\src\main\javascript</sourceDir>

but it means that the configuration can't work on Linux.

**Comment by php-coder** [comment]
> .replace(sourceFileName.toString(), "")

Looks like this line contains a bug: if path to a source file has a segment with a same name, it also will be replaced.

**Comment by milpol** [comment]
Filename replacement is fixed, but frankly I don't have any decent idea how to resolve win-nix portability.
I can do conditional fixing with sys recognition, but that smells. Other option is to support comma separated inputs. That would look strange but could be used to `Path` creation.

**Comment by php-coder** [comment]
> Filename replacement is fixed, but frankly I don't have any decent idea how to resolve win-nix portability.

Yeah, I agree, it's tricky. I would play with converting everything to a Unix-style paths because Windows can understand this. Also, perhaps, [relativization](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#relativize(java.nio.file.Path)) might help?

**Comment by jonaskoperdraat** [comment]
So I've added a PR (#11 and #12) that fixes this issue. Mostly, I've resolved to using Paths wherever possible, and falling back to File where necessary. These work with file system separators, and when provisioning the plugin, Maven seems to do the right conversion when using File parameters. In case of the jsIncludes and jsExcludes parameters, these are converted from Unix-style path separators to filesystem-style path separators (if required), because the library used for scanning files requires the right kind of separators

In order to determine the realtive path, I indeed used relativization 👍
